### ISSUE TITLE
[subplat] Add Firefox Newsletter Opt-in Checkbox for New User Checkout Flow

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#9786]
'mozilla-and-you' is the newsletter.  That need to be included in the request when creating the new user.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-3710)

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

This will likely require multiple small patches, so, at the very least, we can work Payments frontend changes concurrently as a team.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Okay I just spoke with Lauren Niolet and Paul McLanahan re: how to verify the newsletter signup (thank you):

  * Firstly, “mozilla-and-you” is indeed the correct newsletter slug.
  * Assuming we are consistently pointing to the same Basket environment in FxA Settings for the “Email Communications” link as when we post to sign people up to the newsletter, QA can verify that they are subscribed by looking in the “Email Communications” page and seeing that “Firefox News” is checked.
    * basket-stage: basket.allizom.org
      * “Email Communications” page will have domain http://www.mozilla.org ( http://www.mozilla.org|smart-link ) , as data will go to prod CTMS and Acoustic

    * basket-prod: basket.mozilla.org
    * basket-dev: basket-dev.allizom.org
      * “Email Communications” page will have domain http://www-dev.allizom.org ( http://www-dev.allizom.org|smart-link ) , as data will go to stage CTMS

  * Notably, while the first newsletter signup for an email address in other contexts does receive a confirm email, this is not the case for signup through FxA (this is in the Marketing team’s backlog).
    * I’m not 100% on this, but I think this is because FxA users are automatically signed up for the Firefox Accounts Tips newsletter when they create a Firefox Account.
    * Tony Cinotto , Marketing says this isn’t a small lift, but I am looping you in here in case this is something worth prioritizing in the future.

So what I need to do is verify that we are consistently pointing to the same Basket environment in FxA stage (either basket-stage or basket-dev) and similarly for FxA prod (basket-prod).

**Comment by data-sync-user** [comment]
➤ Tony Cinotto commented:

Bianca Danforth just to confirm, a confirm email confirms that you want to receive these newsletter emails or you can progressively opt out? for our case, a user wouldn’t receive this confirm email and would start receiving the newsletter?

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Yes, sorry, this is a bit confusing. I should have used the term “welcome email”, and by that I mean an email acknowledging that the user signed up successfully.

While there is no “welcome email”, there is a “confirm email” which is used in non-FxA cases for new email addresses that require verification. In the FxA case, this “confirm email” isn’t sent, as it is assumed the user’s email address has already been verified.

In other words, currently we should not expect any newsletter emails after successful sign-up. I was trying to point out that we may want to send a “welcome email” in our case in the future.

**Comment by data-sync-user** [comment]
➤ Madalin Cotetiu commented:

Bianca Danforth, we created some steps in order to verify the newsletter signup and we will make sure that we check this.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Thanks Madalin Cotetiu . Not sure if this is in your steps, but the recommended way to check in the absence of an email is to look in FxA Settings > Email Communications that the “Firefox News” checkbox is checked.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

STR

This should be verifiable in Train 213 in stage.

  1. Without being logged into an existing FxA account, click to subscribe to a VPN product/plan.
  2. On the new passwordless checkout page, complete the payment form, being sure to check the newsletter checkbox underneath the second email field before submitting the form.
  3. After seeing the checkout success page, complete the set up process by checking your email and setting a password.
  4. Log in to the FxA account.
  5. Go to FxA Settings and click “Email Communications” on the left sidebar.

Expected results:

  1. The checkbox next to “Firefox News” is checked.
  2. No newsletter email is received to verify the email address; this would be a separate email from the “finish setup” email for the subscription that is referenced in Step 3.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

We verified this in Stage - Train 213 and logged the issue: https://mozilla-hub.atlassian.net/browse/FXA-3865 ( https://mozilla-hub.atlassian.net/browse/FXA-3865|smart-link ) ;
